src/pages/Homepage.jsx

```diff
@@ -1,70 +1,82 @@
-import { useState } from 'react';
+import { useCallback, useState } from 'react';
+import { v4 as uuid } from 'uuid';
 import TimerComponent from '../components/TimerComponent';
 import ActiveTeamsComponent from '../components/ActiveTeamsComponent';
 import EliminatedTeamsComponent from '../components/EliminatedTeamComponent';
+import findTeam from '../utils/index.js';
 
 const Homepage = () => {
   const [teams, setTeams] = useState([]);
   const [draggedIndex, setDraggedIndex] = useState(null);
 
   const handleAddTeam = (type) => {
-    setTeams([
-      ...teams,
+    setTeams((prevState) => [
+      ...prevState,
       {
+        id: uuid(),
         name: '',
         score: 0,
-        active: type === 'active' ? true : false,
+        active: type === 'active',
       },
     ]);
   };
 
-  const handleNameInput = (e, index) => {
+  const handleNameInput = useCallback((event) => {
+    const id = event.target.closest('li').dataset.teamid;
+    const index = findTeam(teams, id);
     const newTeams = [...teams];
-    newTeams[index].name = e.target.value;
+    newTeams[index].name = event.target.value;
     setTeams(newTeams);
-  };
+  }, [setTeams, teams]);
 
-  const handleScoreIncrease = (index) => {
+  const handleScoreIncrease = useCallback((id) => () => {
```

Comment:
This is another way of using useCallback but still be able to pass down some additional data. That's helpful if you're not allowed to add any arbitrary data to the DOM for whatever reason. It's a little bit harder to read at first, but one get used to it.

Please see `handleScoreIncrease` in the components on how to use it.

Reply:
Will do, thank you.